validation/src/main/java/dk/kvalitetsit/itukt/validation/service/ValidationServiceAdaptor.java

```diff
@@ -1,27 +1,69 @@
 
-import dk.kvalitetsit.itukt.common.Mapper;
-import dk.kvalitetsit.itukt.validation.service.model.DataContext;
-import org.openapitools.model.ValidationRequest;
+import dk.kvalitetsit.itukt.validation.service.model.ValidationInput;
+import dk.kvalitetsit.itukt.validation.service.model.ValidationResult;
+import org.openapitools.model.*;
+
+import java.util.List;
 
 /**
  * The {@code ValidationServiceAdaptor} class is responsible for adapting between the boundary layer and the service layer {@link ValidationServiceImpl}.
  * <p>
  * This class accommodates isolation in terms of testing as mocking mappers will be avoided
  */
-public class ValidationServiceAdaptor implements ValidationService<ValidationRequest> {
+public class ValidationServiceAdaptor implements ValidationService<ValidationRequest, ValidationResponse> {
 
-    private final ValidationService<DataContext> service;
-    private final Mapper<ValidationRequest, DataContext> mapper;
+    private final ValidationService<ValidationInput, ValidationResult> validationService;
 
-    public ValidationServiceAdaptor(ValidationService<DataContext> service, Mapper<ValidationRequest, DataContext> mapper) {
-        this.service = service;
-        this.mapper = mapper;
+    public ValidationServiceAdaptor(ValidationService<ValidationInput, ValidationResult> validationService) {
+        this.validationService = validationService;
     }
 
     @Override
-    public boolean validate(ValidationRequest request) {
-        DataContext ctx = mapper.map(request);
-        return service.validate(ctx);
+    public ValidationResponse validate(ValidationRequest request) {
+        var responses = request.getValidate().stream()
+                .map(validate -> validate(request, validate))
+                .toList();
+        return combineResponses(responses);
+    }
+
+    private ValidationResponse validate(ValidationRequest request, Validate validate) {
+        var validationInput = mapToValidationInput(request, validate);
+        var result = validationService.validate(validationInput);
+        return switch (result) {
+            case dk.kvalitetsit.itukt.validation.service.model.ValidationSuccess ignored -> new ValidationSuccess();
+            case dk.kvalitetsit.itukt.validation.service.model.ValidationError error -> mapValidationError(validate, error);
+        };
+    }
+
+    private ValidationResponse combineResponses(List<ValidationResponse> responses) {
+        if (responses.stream().allMatch(resp -> resp instanceof ValidationSuccess)) {
+            return new ValidationSuccess();
+        } else {
+            var validationErrors = responses.stream()
+                    .filter(resp -> resp instanceof ValidationFailed)
+                    .flatMap(failed -> ((ValidationFailed) failed).getValidationErrors().stream())
+                    .toList();
+            return new ValidationFailed()
+                    .validationErrors(validationErrors);
+        }
+    }
```

Comment:
Jeg synes egentlig denne er fin, men tænkte på om cast kunne undgås. Jeg kiggede lidt på det, og kom frem til denne løsning, som også er mere idiomatisk med Stream API'et.

```suggestion
    private ValidationResponse combineResponses(List<ValidationResponse> responses) {
        var validationErrors = responses.stream()
                .flatMap(r -> r instanceof ValidationFailed f ? f.getValidationErrors().stream() : Stream.of())
                .toList();
        return validationErrors.isEmpty() ?
                new ValidationSuccess() :
                new ValidationFailed().validationErrors(validationErrors);
    }
```

Reply:
Jeg tog dit andet forslag